apidef/api_definitions.go

```diff
@@ -540,6 +540,14 @@ func (s *StringRegexMap) Check(value string) string {
 	return s.matchRegex.FindString(value)
 }
 
+func (s *StringRegexMap) FindStringSubmatch(value string) []string {
+	return s.matchRegex.FindStringSubmatch(value)
+}
+
+func (s *StringRegexMap) FindAllStringSubmatch(value string, n int) [][]string {
+	return s.matchRegex.FindAllStringSubmatch(value, n)
```

Comment:
the same thing for `FindAllStringSubmatch ` we need to add caching logic for this function in `tyk/regexp` so we won't spend time on RX with the same params

Reply:
Looks like it's already done
https://github.com/TykTechnologies/tyk/blob/23e56e71ddfa5beb0f50a2aebe32e2ebccb3e68e/regexp/regexp.go#L402

Reply:
ah, right. this one is already cached